eventstore/tasks.py

```diff
@@ -319,6 +323,76 @@ def update_rapidpro_contact_error_timestamp(context):
     )
 
 
+@app.task(
+    autoretry_for=(RequestException, ConnectionException),
+    retry_backoff=True,
+    max_retries=15,
+    acks_late=True,
+    soft_time_limit=10,
+    time_limit=15,
+)
+def upload_and_delete_cdu_address_updates(self):
+
+    Headers = [
+        "msisdn",
+        "first_name",
+        "last_name",
+        "id_type",
+        "id_number",
+        "date_of_birth",
+        "folder_number",
+        "district",
+        "municipality",
+        "city",
+        "suburb",
+        "street_name",
+        "street_number",
+    ]
+    data = []
+    ids_to_delete = []
+    for address_update in CDUAddressUpdate.objects.all():
+        ids_to_delete.append(address_update.id)
+        row = ",".join([str(getattr(address_update, header)) for header in Headers])
+        data.append(row)
+
+    # Write to csv
+    with open("./CDU_address.csv", "w", newline="") as csv_file:
+        writer = csv.writer(csv_file)
+        writer.writerow(Headers)
+        for row in data:
+            writer.writerow(row.replace('"', "").split(","))
+
+    # Start FTP Session
+    cnopts = pysftp.CnOpts()
+    cnopts.hostkeys = None
```

Comment:
This will cause us to be susceptible to man-in-the-middle attacks. Can we set the key for the server in the settings file, and use it here?

That will mean that it will break if the server key ever changes, but that's exactly what we want, because that means that we're no longer talking to the same server.

There's a nice guide here to setting the key from a string: https://stackoverflow.com/questions/38939454/verify-host-key-with-pysftp/43389508#43389508

Reply:
Thanks. I had seen that and was thinking the same thing but wouldn't that mean that we have to get their public key and use that as the value of cnopts.hostkeys

Reply:
Yes, we would need their public key. We could get it using `ssh-keyscan`